language-server/modules/langserver-core/src/main/java/org/ballerinalang/langserver/common/utils/FilterUtils.java

```diff
@@ -201,20 +206,17 @@ private static PackageID getPackageIDForBType(BType bType) {
         return bType.tsymbol.pkgID;
     }
     
-    private static String getBTypeForUnionType(BType bType) {
-        if (!(bType instanceof BUnionType)) {
-            return "";
-        }
-        List<BType> memberTypeList = new ArrayList<>(((BUnionType) bType).getMemberTypes());
+    private static BType getBTypeForUnionType(BUnionType bType) {
+        List<BType> memberTypeList = new ArrayList<>(bType.getMemberTypes());
```

Comment:
Shall we keep this as a `Set<BType>` since  `bType.getMemberTypes()` returns a **Set**. Later we can get the first element by;
```
if (memberTypeList.size() == 1) {
    return memberTypeList.iterator().next().toString();
}
```

Reply:
Since the item removal affects the original member set, shall we copy those to a list on behalf of processing?

Reply:
if the requirement is to keeping the original set. +1 to keep it as it is.